src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -26,13 +26,34 @@ internal unsafe partial class CorInfoImpl
         [DllImport("ryujit")]
         private extern static IntPtr getJit();
 
+        [DllImport("jitinterface")]
+        private extern static IntPtr GetJitInterfaceWrapper(IntPtr unwrapped);
+
         private IntPtr _jit;
 
-        [UnmanagedFunctionPointerAttribute(CallingConvention.StdCall)]
-        private delegate CorJitResult _compileMethod(IntPtr _this, IntPtr comp, ref CORINFO_METHOD_INFO info, uint flags,
+        [DllImport("jitinterface")]
+        private extern static CorJitResult JitWrapper(out IntPtr exception, IntPtr _this, IntPtr comp, ref CORINFO_METHOD_INFO info, uint flags,
             out IntPtr nativeEntry, out uint codeSize);
 
-        private _compileMethod _compile;
+        [DllImport("jitinterface")]
+        private extern static IntPtr AllocException([MarshalAs(UnmanagedType.LPWStr)]string message, int messageLength);
+
+        private IntPtr AllocException(Exception ex)
+        {
+            string exString = ex.ToString();
+            IntPtr nativeException = AllocException(exString, exString.Length);
+            lock (_nativeExceptions)
```

Comment:
This lock is not necessary - we assume in many places that the given CorInfoImpl instance is used from a single thread only.